packages/ember-toucan-form/src/-private/combobox-field.ts

```diff
@@ -0,0 +1,72 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+import type { HeadlessFormBlock, UserData } from './types';
+import type {
+  Option,
+  ToucanFormComboboxFieldComponentSignature as BaseComboboxFieldSignature,
+} from '@crowdstrike/ember-toucan-core/components/form/fields/combobox';
+import type { FormData, FormKey, ValidationError } from 'ember-headless-form';
+
+export interface ToucanFormComboboxFieldComponentSignature<
+  DATA extends UserData,
+  KEY extends FormKey<FormData<DATA>> = FormKey<FormData<DATA>>
+> {
+  Element: HTMLInputElement;
+  Args: Omit<
+    BaseComboboxFieldSignature<Option>['Args'],
+    'error' | 'onChange'
+  > & {
+    /**
+     * The name of your field, which must match a property of the `@data` passed to the form
+     */
+    name: KEY;
+
+    /*
+     * @internal
+     */
+    form: HeadlessFormBlock<DATA>;
+  };
+  // TODO: How do we get this to play nicely with our
+  // generic in toucan-core?
+  // `BaseComboboxFieldSignature<Option>['Blocks'];`
+  // gives a glint error!
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Blocks: BaseComboboxFieldSignature<any>['Blocks'];
```

Comment:
I couldn't get this to work as expected unfortunately so I made the type `any`.  We want the blocks to take a generic like the Combobox does, but since this component is already generic due to the headless-form data, I'm not sure how to make this match the typing of `@options` like we do above! Any thoughts?  Maybe @simonihmig ?

To reproduce you can pull this repo down and the test at https://github.com/CrowdStrike/ember-toucan-core/pull/200/files#diff-a7fdb0c7554c73ada47e314a1462054c5557c84b0bdf6649337495dc0cc96c5eR31 should throw an error if you change this from `any` to something else.

Reply:
I think we would have to introduce `OPTION` here also, and pass it from `ToucanFormComboboxFieldComponent` to `ToucanFormComboboxFieldComponentSignature` down to `BaseComboboxFieldSignature`. have you tried that?